code/__DEFINES/traits.dm

```diff
@@ -114,6 +114,30 @@
 #define HAS_TRAIT_FROM_ONLY(target, trait, source) (target.status_traits?[trait] && (source in target.status_traits[trait]) && (length(target.status_traits[trait]) == 1))
 #define HAS_TRAIT_NOT_FROM(target, trait, source) (target.status_traits?[trait] && (length(target.status_traits[trait] - source) > 0))
 
+/// Define for adding multiple traits to a target via a list. Must have a common source and target.
+#define ADD_TRAIT_LIST(target, list_of_traits, source) \
+	do { \
+		if (!islist(list_of_traits)) { \
+			stack_trace("Invalid arguments passed to ADD_TRAIT_LIST! Invoked on [target] with [list_of_traits], source being [source]."); \
+			return; \
+		} \
```

Comment:
1. While this is a macro, this definitely should not `return` since that stops the entire proc its in
2. You should use `ASSERT` instead--`ASSERT(islist(list_of_traits), "my message here")`